api/test/e2e/journeys/test_schemas.py

```diff
@@ -52,23 +45,116 @@ def update_schema(self, schema: dict) -> Response:
             json=schema,
         )
 
+    def get_dataset_from_dynamodb(self, layer: str, domain: str, dataset: str) -> dict:
+        dynamodb = boto3.resource("dynamodb", region_name=AWS_REGION)
+        table = dynamodb.Table(SCHEMA_TABLE_NAME)
+        try:
+            response = table.query(
+                KeyConditionExpression=Key("PK").eq(f"{layer}/{domain}/{dataset}"),
+            )
+        except ClientError as e:
+            raise e
+        return response["Items"]
+
+    def extract_column_names(self, dataset_schema: dict) -> list:
+        return [item["name"] for item in dataset_schema[0]["columns"]]
+
     def test_uploads_new_schema_version(self):
         # Upload initial schema
         schema_v1 = self.read_schema_version(SchemaVersion.V1)
+        layer = schema_v1["metadata"]["layer"]
+        domain = schema_v1["metadata"]["domain"]
+        dataset = schema_v1["metadata"]["dataset"]
+        column_names = [column["name"] for column in schema_v1["columns"]]
         response1 = self.upload_schema(schema_v1)
 
         assert response1.status_code == HTTPStatus.CREATED
 
-        # TODO: Later - Verify that initial schema is correct
-        # res = requests.get(
-        #     self.info_dataset_url(self.layer, self.e2e_test_domain, self.dataset),
-        #     headers=self.generate_auth_headers(),
-        # )
-        # data = res.json()
-        # assert data == schema_v1
+        # Verify schema in dynamodb is correct -> No way to query schema via API endpoint
+        dataset_schema_v1 = self.get_dataset_from_dynamodb(layer, domain, dataset)
+        rapid_schema_column_names_v1 = self.extract_column_names(dataset_schema_v1)
+        assert dataset_schema_v1[0]["domain"] == schema_v1["metadata"]["domain"]
+        assert dataset_schema_v1[0]["layer"] == schema_v1["metadata"]["layer"]
+        assert sorted(column_names) == sorted(rapid_schema_column_names_v1)
```

Comment:
Can you just assert the whole thing rather than having 3 separate ones for the domain, layer and columns?